### DISCUSSION TITLE
Raw value while adding items to M2M and O2M relations differs

### DISCUSSION THREAD

**Discussion by viters** [directus/directus#16363] [Archive]
**Describe the Bug**

On M2M, if you add completely new relation, it goes to `create` array in `raw value`. In O2M on the other hand, it goes to `update`:

![image](https://user-images.githubusercontent.com/13100466/200641955-8d432f07-1f6a-4c93-b249-262c5a085d76.png)

_Originally posted by @viters in https://github.com/directus/directus/issues/16291#issuecomment-1307630005_

**What version of Directus are you using?**
9.20.3

**What version of Node.js are you using?**
16.18.0

**Comment by viters** [comment]
I think that interface is kinda unintuitive. There are two interchanging meanings of `create`, `update` and `delete`:
- theres "relational" meaning, mostly visible in M2M but also in O2M while removing relation, the keywords are referring to relation itself - in o2m for example if you remove relation, it goes to `delete` array
- theres "target" meaning, in O2M if you create relation via `create new` it goes to `create`, and if you create relation via `use existing` it goes to `update` (as it updates the target element)

Target meaning is different for M2M relations, as `target` is junction table item, not the actual relation target.

**Reply by rijkvanzanten** [reply]
>  There are two interchanging meanings of create, update and delete:

Depends on how you look at it. `create` always creates a new record, and `update` always updates an existing record, so no matter the GUI view that sets those values, the underlying calls behave the same. A M2M is just a O2M to a junction table, with a M2O from the junction to the "target". The GUI abstracts that away into a list of targets, but the raw values work the same 👍🏻
